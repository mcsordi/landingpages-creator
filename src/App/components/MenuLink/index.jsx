import P from 'prop-types';

function MenuLink({ text, link, newTab = false }) {
  return (
    <div>
      <a
        className="relative text-primary m-2 py-2 text-xsmall hvr-underline-from-center"
        href={link}
        target={newTab ? '_blank' : '_self'}
        rel="noreferrer"
      >
        {text}
      </a>
    </div>
  );
}
MenuLink.propTypes = {
  link: P.string.isRequired,
  newTab: P.bool,
  text: P.string.isRequired,
};
export default MenuLink;
