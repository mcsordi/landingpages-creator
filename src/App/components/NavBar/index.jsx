import { SlMenu } from 'react-icons/sl';
import { IoClose } from 'react-icons/io5';
import P from 'prop-types';
import { useState } from 'react';

function NavBar({ children }) {
  const [display, setDisplay] = useState(false);
  return (
    <>
      <nav
        className={`${display ? 'flex' : 'hidden'}   items-center justify-evenly flex-col
        min-h-screen bg-white w-full py-1.5 md:flex md:min-h-0 md:flex-row`}
      >
        {children}
      </nav>
      <div className={`flex fixed md:hidden mx-auto right-0 p-4`}>
        <div className="bg-[#0A1128] p-1.5 text-white text-medium cursor-pointer">
          {display ? (
            <IoClose
              onClick={() => {
                setDisplay(false);
              }}
            />
          ) : (
            <SlMenu
              onClick={() => {
                setDisplay(true);
              }}
            />
          )}
        </div>
      </div>
    </>
  );
}
NavBar.propTypes = {
  children: P.node,
  display: P.bool,
};
export default NavBar;
