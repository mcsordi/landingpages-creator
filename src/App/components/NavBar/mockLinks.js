export default [
  {
    text: 'texto1',
    link: '#texto1',
  },
  {
    text: 'texto2',
    link: '#texto2',
  },
  {
    text: 'texto3',
    link: '#texto3',
  },
  {
    text: 'texto4',
    link: '#texto4',
  },
  {
    text: 'texto5',
    link: '#texto5',
  },
  {
    text: 'texto6',
    link: '#texto6',
  },
  {
    text: 'texto7',
    link: '#texto7',
  },
  {
    text: 'texto8',
    link: '#texto8',
  },
  {
    text: 'texto9',
    link: '#texto9',
  },
  {
    text: 'texto10',
    link: '#texto10',
  },
];
