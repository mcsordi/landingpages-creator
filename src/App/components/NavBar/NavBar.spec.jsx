import { render, screen } from '@testing-library/react';
import NavBar from '.';
import MenuLink from '../MenuLink';
describe('<NavBar/>', () => {
  it('should render NavBar Component', () => {
    render(
      <NavBar>
        <MenuLink link={'https://www.google.com.br'} text={'google'} />
      </NavBar>,
    );
    const component = screen.getByRole('link');
  });
});
