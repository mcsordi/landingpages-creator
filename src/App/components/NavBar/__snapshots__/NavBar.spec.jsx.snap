// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar/> should match a snapshot 1`] = `
<div>
  <nav
    class="hidden fixed z-10  items-center justify-evenly flex-col
            min-h-screen bg-white w-full py-1.5 md:flex md:min-h-0 md:flex-row"
  />
  <div
    class="flex fixed md:hidden mx-auto right-0 p-4 z-30"
  >
    <div
      class="bg-[#0A1128] p-1.5 text-white text-medium cursor-pointer"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M27 193.6c-8.2-8.2-12.2-18.6-12.2-31.2s4-23 12.2-31.2S45.6 119 58.2 119h912.4c12.6 0 23 4 31.2 12.2s12.2 18.6 12.2 31.2-4 23-12.2 31.2-18.6 12.2-31.2 12.2H58.2c-12.6 0-23-4-31.2-12.2zm974.8 285.2c8.2 8.2 12.2 18.6 12.2 31.2s-4 23-12.2 31.2-18.6 12.2-31.2 12.2H58.2c-12.6 0-23-4-31.2-12.2S14.8 522.6 14.8 510s4-23 12.2-31.2 18.6-12.2 31.2-12.2h912.4c12.6 0 23 4 31.2 12.2zm0 347.4c8.2 8.2 12.2 18.6 12.2 31.2s-4 23-12.2 31.2-18.6 12.2-31.2 12.2H58.2c-12.6 0-23-4-31.2-12.2S14.8 870 14.8 857.4s4-23 12.2-31.2S45.6 814 58.2 814h912.4c12.6 0 23 4.2 31.2 12.2z"
        />
      </svg>
    </div>
  </div>
</div>
`;
